script.js

```diff
@@ -1,4 +1,492 @@
+      } else if (dealerHand[0].value > 21) {
+        players[i].hands[handCounter].roundResult = WIN;
+        // Player loses if their hand value is lower than or equal to dealer's
+      } else if (findOptimalValue(players[i].hands[handCounter]) <= findOptimalValue(dealerHand[0])) {
+        players[i].hands[handCounter].roundResult = LOSE;
+        // Player wins if their hand value is higher than dealer's
+      } else if (findOptimalValue(players[i].hands[handCounter]) > findOptimalValue(dealerHand[0])) {
+        players[i].hands[handCounter].roundResult = WIN;
+      }
+      // Payout the winning bets
+      if (players[i].hands[handCounter].roundResult == WIN) {
+        players[i].chips += 2 * players[i].hands[handCounter].playerBet; // Winning player gets back their original bet + the winnings
+      }
+    }
+  }
+  // Display the results of the round
+  var roundResults = displayRoundResults(players);
+  // Remove the players who have 0 chips remaining at the end of the round
+  roundResults += eliminatePlayers(players);
+  // If there are still players remaining that can play the next round
+  if (players.length > 0) {
+    turn = 0; // Reset the turn counter for the next round
+    // Reset the hands for all players and the dealer for the next round
+    for (var i = 0; i < players.length; i++) {
+      players[i].hands = [{ cards: [], value: 0, handNum: 1, playerBet: 0 }];
+    }
+    dealerHand = [{ cards: [], value: 0, handNum: 1 }];
+    // Reset the deck for the next round
+    deck = [];
+    gameMode = PLACE_BETS; // Update game mode for the next round
+    // Prompt players to start the next round
+    roundResults += `<br><br>Please enter your bet for the next round, ${players[turn].playerName}.`;
+  } else {
+    // There are no more eligible players, so the game ends
+    roundResults += `<br><br>${noPlayersLeft()}`;
+    gameMode = NO_PLAYERS_LEFT;
+  }
+  return roundResults;
+};
+
+// Game mode where we tell the user that there are no eligible players remaining
+var noPlayersLeft = function () {
+  var resultMessage = `All players have been eliminated 😭<br><br>Refresh the page to play again.`;
+  return resultMessage;
+};
+
+// Helper function to remove players with 0 chips remaining, as they will not be able to play the next round
+var eliminatePlayers = function (playersArray) {
+  var resultMessage = "";
+  // Check if there are any players to be removed
+  if (playersArray.find(({ chips }) => chips == 0)) {
+    var eliminatedPlayers = [];
+    // When removing elements from an array, we need to loop backwards as we are modifying the length of the array, so as to not skip any elements
+    for (var i = playersArray.length - 1; i >= 0; i--) {
+      if (playersArray[i].chips == 0) {
+        // Remove players with 0 chips and add them into the eliminatedPlayers array
+        eliminatedPlayers = [...eliminatedPlayers, ...playersArray.splice(i, 1)];
+      }
+    }
```

Comment:
It is generally not good practice to modify the thing in which the for loop depends on. It would be better to create a variable called length and use that as your condition in your for loop. Thereafter, you would be free to modify the playerArray in any way without being afraid of the code breaking